src/CoreWCF.RabbitMQ/src/CoreWCF/Channels/RabbitMqBinding.cs

```diff
@@ -2,89 +2,68 @@
-        /// Protocols.DefaultProtocol.
+        /// Specifies the version of the AMQP protocol that should be used to communicate with the broker
         /// </summary>
-        /// <param name="hostname">The hostname of the broker to connect to</param>
-        /// <param name="port">The port of the broker to connect to</param>
-        public RabbitMqBinding(string hostname, int port)
-            : this(hostname, port, Protocols.DefaultProtocol)
-        {
-        }
+        public IProtocol BrokerProtocol { get; set; }
+        public SslOption SslOption { get; set; }
+        public string VirtualHost { get; set; }
 
         /// <summary>
-        /// Uses the broker and protocol specified
+        /// Credentials to access the RabbitMQ host
         /// </summary>
-        /// <param name="hostname">The hostname of the broker to connect to</param>
-        /// <param name="port">The port of the broker to connect to</param>
-        /// <param name="protocol">The protocol version to use</param>
-        public RabbitMqBinding(string hostname, int port, IProtocol protocol)
-            : this(protocol)
-        {
-            HostName = hostname;
-            Port = port;
-        }
+        public ICredentials Credentials { get; set; }
 
         /// <summary>
-        /// Uses the broker, login and protocol specified
+        /// Configuration used for declaring a queue
         /// </summary>
-        /// <param name="hostname">The hostname of the broker to connect to</param>
-        /// <param name="port">The port of the broker to connect to</param>
-        /// <param name="username">The broker username to connect with</param>
-        /// <param name="password">The broker password to connect with</param>
-        /// <param name="virtualHost">The broker virtual host</param>
-        /// <param name="maxMessageSize">The largest allowable encoded message size</param>
-        /// <param name="protocol">The protocol version to use</param>
-        public RabbitMqBinding(string hostname, int port, string username, string password, string virtualHost,
-            long maxMessageSize, IProtocol protocol)
-            : this(protocol)
-        {
-            HostName = hostname;
-            Port = port;
-            _transport.Credentials = new NetworkCredential(username, password);
-            _transport.VirtualHost = virtualHost;
-            MaxMessageSize = maxMessageSize;
-        }
+        public QueueDeclareConfiguration QueueConfiguration { get; set; }
 
         /// <summary>
-        /// Uses the specified protocol. The broker must be set before use.
+        /// Gets the scheme used by the binding
         /// </summary>
-        /// <param name="protocol">The protocol version to use</param>
-        public RabbitMqBinding(IProtocol protocol)
+        public override string Scheme
+        {
+            get { return CurrentVersion.Scheme; }
+        }
```

Comment:
It's typical for a binding to defer to the transport for the scheme. So this would become:
```c#
public override string Scheme => _transport.Scheme;
```